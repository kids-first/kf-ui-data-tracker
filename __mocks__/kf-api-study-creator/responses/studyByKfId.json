{
  "data": {
    "studyByKfId": {
      "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
      "name": "benchmark extensible e-business",
      "shortName": null,
      "bucket": "stay-wrong-ready",
      "kfId": "SD_8WX8QQ06",
      "createdAt": "2019-04-20T15:22:18.780221+00:00",
      "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
      "attribution": null,
      "dataAccessAuthority": "dbGaP",
      "externalId": "test_study_external_id",
      "releaseStatus": null,
      "version": null,
      "releaseDate": "2019-08-24",
      "description": "Sudy description in markdown, commonly the X01 abstract text. --  As test study description",
      "anticipatedSamples": 100,
      "awardeeOrganization": "CHOP",
      "collaborators": {
        "edges": [
          {
            "node": {
              "id": "VXNlck5vZGU6NQ==",
              "username": "Jennifer Warren",
              "firstName": "Marc",
              "lastName": "Wilson",
              "email": "ericksontracy@benitez-ruiz.com",
              "picture": "",
              "roles": [],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          },
          {
            "node": {
              "id": "VXNlck5vZGU6OQ==",
              "username": "Anthony Hamilton",
              "firstName": "John",
              "lastName": "Garza",
              "email": "richard06@merritt.net",
              "picture": "",
              "roles": [],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          },
          {
            "node": {
              "id": "VXNlck5vZGU6MjY=",
              "username": "devadmin",
              "firstName": "",
              "lastName": "",
              "email": "admin@myproject.com",
              "picture": "",
              "roles": ["ADMIN"],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          }
        ],
        "__typename": "UserNodeConnection"
      },
      "files": {
        "edges": [
          {
            "node": {
              "id": "RmlsZU5vZGU6U0ZfNVpQRU0xNjc=",
              "kfId": "SF_5ZPEM167",
              "name": "organization.jpeg",
              "fileType": "CLN",
              "tags": [],
              "description": "Month necessary animal end standard case. View system operation message decade. Actually sing because deal everything woman subject.",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167",
              "versions": {
                "edges": [
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl80TlpOQkVPRg==",
                      "kfId": "FV_NVV741M1",
                      "state": "PEN",
                      "size": 9893,
                      "createdAt": "2018-08-21T03:16:11+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl83NkhONTdaNw==",
                      "kfId": "FV_KCDKCDYR",
                      "state": "PEN",
                      "size": 5460,
                      "createdAt": "2018-09-27T05:32:26+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9IUkFHWE5XMQ==",
                      "kfId": "FV_800JI8IP",
                      "state": "PEN",
                      "size": 5419,
                      "createdAt": "2017-08-25T00:05:30+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  }
                ],
                "__typename": "VersionNodeConnection"
              },
              "__typename": "FileNode"
            },
            "__typename": "FileNodeEdge"
          },
          {
            "node": {
              "id": "RmlsZU5vZGU6U0ZfWTA3SU4xSE8=",
              "kfId": "SF_Y07IN1HO",
              "name": "its.mp3",
              "fileType": "SHM",
              "tags": ["dbGaP", "random", "tag that has ten more char"],
              "description": "Question meeting move recognize.",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_Y07IN1HO",
              "versions": {
                "edges": [
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9UWjVROEhYWA==",
                      "kfId": "FV_IJ2CN4P9",
                      "state": "PEN",
                      "size": 7394,
                      "createdAt": "2017-08-22T08:32:38+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9aOUM1SUUwRg==",
                      "kfId": "FV_DNTN03CS",
                      "state": "PEN",
                      "size": 5883,
                      "createdAt": "2017-09-02T21:02:35+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9ITDg0SDhHTw==",
                      "kfId": "FV_ZDB8JOCT",
                      "state": "PEN",
                      "size": 1899,
                      "createdAt": "2019-03-14T05:46:35+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9YRVg4TERHMw==",
                      "kfId": "FV_10SBI6VA",
                      "state": "PEN",
                      "size": 7631,
                      "createdAt": "2018-03-13T23:52:08+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9HV1JLUUVDQQ==",
                      "kfId": "FV_8KF1VMN9",
                      "state": "PEN",
                      "size": 6834,
                      "createdAt": "2019-04-06T07:51:17+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  }
                ],
                "__typename": "VersionNodeConnection"
              },
              "__typename": "FileNode"
            },
            "__typename": "FileNodeEdge"
          }
        ],
        "__typename": "FileNodeConnection"
      },
      "events": {
        "edges": [
          {
            "node": {
              "id": "RXZlbnROb2RlOjM5",
              "uuid": "b790803e-580e-458c-8b42-78252a5403f6",
              "createdAt": "2019-08-26T18:25:02.985155+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_9AF7CC3Z of file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM4",
              "uuid": "6372f6f7-8556-4b1c-9c29-7c0a7923f7ca",
              "createdAt": "2019-08-26T18:24:46.341665+00:00",
              "eventType": "FV_UPD",
              "description": "devadmin updated version FV_4FCEYZ3T of file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM2",
              "uuid": "ee949b92-ddb0-491b-b807-8f754e1641ff",
              "createdAt": "2019-08-26T18:24:36.763314+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_ABCPEQRK of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM1",
              "uuid": "05379686-3d68-4fed-b6a5-a5e47d97996f",
              "createdAt": "2019-08-26T18:24:19.986588+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_BHXN2F7A of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM0",
              "uuid": "1ec8bb79-b6ad-47eb-bd40-7ffc81135abe",
              "createdAt": "2019-08-26T18:24:03.184513+00:00",
              "eventType": "FV_UPD",
              "description": "devadmin updated version FV_Z0NZQZNX of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMz",
              "uuid": "b9b6bf82-1bdc-4d79-b69b-5e0350d64e52",
              "createdAt": "2019-08-26T18:24:03.182570+00:00",
              "eventType": "SF_UPD",
              "description": "devadmin updated file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMy",
              "uuid": "02564a4f-ead8-4ebb-af12-7171dcc05fd4",
              "createdAt": "2019-08-26T18:23:53.703031+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_4FCEYZ3T of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMx",
              "uuid": "578c24b2-dcf3-4a80-a4e8-d240e9549736",
              "createdAt": "2019-08-26T18:23:53.695244+00:00",
              "eventType": "SF_CRE",
              "description": "devadmin created file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          }
        ],
        "__typename": "EventNodeConnection"
      },
      "projects": {
        "edges": [
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmY=",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:41+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:41+00:00",
              "name": "test study name",
              "projectId": "zhux3/sd-8wx8qq06",
              "projectType": "DEL",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06",
              "workflowType": null,
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          },
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmYtYndhLW1lbQ==",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:42+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:42+00:00",
              "name": "test study name bwa-mem",
              "projectId": "zhux3/sd-8wx8qq06-bwa-mem",
              "projectType": "HAR",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06-bwa-mem",
              "workflowType": "bwa_mem",
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          },
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmYtZ2F0ay1oYXBsb3R5cGVjYWxsZXI=",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:43+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:43+00:00",
              "name": "test study name gatk-haplotypecaller",
              "projectId": "zhux3/sd-8wx8qq06-gatk-haplotypecaller",
              "projectType": "HAR",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06-gatk-haplotypecaller",
              "workflowType": "gatk_haplotypecaller",
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          }
        ],
        "__typename": "ProjectNodeConnection"
      },
      "__typename": "StudyNode"
    }
  }
}
