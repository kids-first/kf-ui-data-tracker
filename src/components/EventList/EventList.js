import {Button, Grid, Icon, List, Popup} from 'semantic-ui-react';
import React, {Fragment} from 'react';

import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';
import TimeAgo from 'react-timeago';
import {eventType} from '../../common/enums';
import {longDate} from '../../common/dateUtils';

/**
 * Highlights any kf_ids in the description and turns them into links
 * If the study, file, or version does not exist, no highlighting will be
 * applied.
 */
const linkDescription = node => {
  // If no study, it must have been deleted including any files or versions
  if (!node.study) return node.description;

  const ids = {
    SD: node.study && node.study.kfId,
    SF: node.file && node.file.kfId,
    FV: node.version && node.version.kfId,
  };

  const urls = {
    SD: `/study/${ids.SD}/basic-info/info`,
    SF: `/study/${ids.SD}/documents/${ids.SF}`,
    FV: `/study/${ids.SD}/documents/${ids.SF}`,
  };

  const EntityLink = ({entityType}) => (
    <Link to={urls[entityType]} key={ids[entityType]}>
      {ids[entityType] + ' '}
    </Link>
  );

  const entityPattern = /((SD|SF|FV)_\w{8})/g;
  const matches = node.description.split(entityPattern);
  var parts = [];
  for (var i = 0; i < matches.length; i += 3) {
    const link =
      matches[i + 2] in ids && ids[matches[i + 2]] ? (
        <EntityLink key={i + matches[i + 2]} entityType={matches[i + 2]} />
      ) : (
        matches[i + 1]
      );
    parts = [...parts, matches[i], link];
  }

  return parts;
};

/**
 * Event list is used to display user activities
 * Consisted with three parts:
 * - Color-coded icon indication action type
 * - Text descrbing who did what
 * - Timeago text indication when the action happened
 */
const EventList = ({events, referralTokenData}) => {
  const StudyPopup = ({event}) => {
    var token = [];
    if (event.eventType === 'RT_CRE') {
      token =
        referralTokenData.length > 0 &&
        referralTokenData.filter(
          ({node}) =>
            node.createdBy &&
            event.user &&
            node.createdBy.id === event.user.id &&
            node.email ===
              event.description.match(
                /([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+)/gi,
              )[0] &&
            node.createdAt.split('T')[0] === event.createdAt.split('T')[0],
        );
    }
    if (event.eventType === 'RT_CLA') {
      token =
        referralTokenData.length > 0 &&
        referralTokenData.filter(
          ({node}) =>
            node.claimedBy && event.user && node.claimedBy.id === event.user.id,
        );
    }
    var splitList = [[], [], []];
    if (token.length > 0 && token[0].node.studies.edges.length > 0) {
      const n = 3;
      const wordsPerLine = Math.ceil(token[0].node.studies.edges.length / 3);
      for (let line = 0; line < n; line++) {
        for (let i = 0; i < wordsPerLine; i++) {
          const value = token[0].node.studies.edges[i + line * wordsPerLine];
          if (!value) continue;
          splitList[line].push(value);
        }
      }
    }

    return (
      <Popup
        wide="very"
        disabled={
          token.length === 0 || token[0].node.studies.edges.length === 0
        }
        trigger={
          <Button
            labelPosition="left"
            className={
              token.length > 0 && token[0].node.studies.edges.length > 0
                ? 'text-primary text-normal text-14'
                : 'text-black text-normal text-14'
            }
          >
            {token.length > 0 && token[0].node.studies.edges.length > 0
              ? token[0].node.studies.edges.length +
                (token[0].node.studies.edges.length > 1 ? ' studies' : ' study')
              : 'studies'}
          </Button>
        }
        content={
          <Grid columns={3}>
            {[0, 1, 2].map(i => (
              <Fragment key={i}>
                {splitList[i].length > 0 &&
                  splitList[i].map(({node}) => (
                    <Grid.Column
                      className="pt-5 pb-5"
                      key={node.kfId}
                      as={Link}
                      target="_blank"
                      to={`/study/${node.kfId}/basic-info/info`}
                    >
                      {node.kfId}
                    </Grid.Column>
                  ))}
              </Fragment>
            ))}
          </Grid>
        }
        on={['click']}
      />
    );
  };

  return (
    <List relaxed="very">
      {events &&
        events.length > 0 &&
        events.map(({node}) => (
          <List.Item key={node.id} data-testid="event-item">
            <Icon
              name={
                node.eventType in eventType
                  ? eventType[node.eventType].iconName
                  : eventType.OTH.iconName
              }
              color={
                node.eventType in eventType
                  ? eventType[node.eventType].iconColor
                  : eventType.OTH.iconColor
              }
            />
            <List.Content>
              <List.Content floated="right">
                <TimeAgo
                  live={false}
                  date={node.createdAt}
                  title={longDate(node.createdAt)}
                />
              </List.Content>

              {node.eventType === 'RT_CRE' || node.eventType === 'RT_CLA' ? (
                <>
                  {node.description.split(' to ')[0] + ' to '}
                  <StudyPopup event={node} />
                </>
              ) : (
                <>{linkDescription(node)}</>
              )}
            </List.Content>
          </List.Item>
        ))}
    </List>
  );
};

EventList.propTypes = {
  /** Array of event object*/
  eventList: PropTypes.array,
};

EventList.defaultProps = {
  eventList: [],
};

export default EventList;
