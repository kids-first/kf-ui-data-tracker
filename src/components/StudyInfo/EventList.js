import React from 'react';
import {List, Icon} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
import {eventType} from '../../common/fileUtils';

const EventList = ({events}) => (
  <List relaxed="very">
    {events.map(({node}) => (
      <List.Item key={node.id}>
        <Icon
          name={eventType[node.eventType].iconName}
          color={eventType[node.eventType].iconColor}
        />
        <List.Content>
          <List.Content floated="right">
            <TimeAgo live={false} date={node.createdAt} />
          </List.Content>
          {node.description}
        </List.Content>
      </List.Item>
    ))}
  </List>
);

export default EventList;
