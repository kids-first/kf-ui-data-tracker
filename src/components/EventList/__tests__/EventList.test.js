import React from 'react';
import {render} from 'react-testing-library';
import studyByKfId from '../../../../__mocks__/kf-api-study-creator/responses/studyByKfId';
import EventList from '../EventList';

it('Event list renders correctly', () => {
  const events = studyByKfId.data.studyByKfId.events.edges;
  const tree = render(<EventList events={events} />);
  expect(tree.container).toMatchSnapshot();
});
