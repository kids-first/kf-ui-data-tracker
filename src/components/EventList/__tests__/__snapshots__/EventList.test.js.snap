// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event list renders correctly -- with mock data 1`] = `
<div>
  <div
    class="ui very relaxed list"
    role="list"
  >
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="green add icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:25:02.985Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin created version 
        FV_9AF7CC3Z
         of file 
        SF_5ZPEM167
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="yellow refresh icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:24:46.341Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin updated version 
        FV_4FCEYZ3T
         of file 
        SF_5ZPEM167
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="green add icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:24:36.763Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin created version 
        FV_ABCPEQRK
         of file 
        SF_Y07IN1HO
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="green add icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:24:19.986Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin created version 
        FV_BHXN2F7A
         of file 
        SF_Y07IN1HO
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="yellow refresh icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:24:03.184Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin updated version 
        FV_Z0NZQZNX
         of file 
        SF_Y07IN1HO
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="yellow refresh icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:24:03.182Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin updated file 
        SF_5ZPEM167
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="green add icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:23:53.703Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin created version 
        FV_4FCEYZ3T
         of file 
        SF_Y07IN1HO
        
      </div>
    </div>
    <div
      class="item"
      data-testid="event-item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="green add icon"
      />
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <time
            datetime="2019-08-26T18:23:53.695Z"
            title="26 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        devadmin created file 
        SF_Y07IN1HO
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Event list renders correctly -- with no data 1`] = `
<div>
  <div
    class="ui very relaxed list"
    role="list"
  />
</div>
`;
